asf_heat_pump_suitability/pipeline/run_scripts/run_add_features.py

```diff
@@ -76,28 +81,48 @@ def parse_arguments() -> argparse.Namespace:
 
 if __name__ == "__main__":
     args = parse_arguments()
-    epc_path = args.epc_path
+    epc_path = args.epc
     save_as = args.save_as
     year = args.year
     q = args.quarter
 
     # Import processed EPC
     logging.info(f"Loading EPC file from path: {epc_path}")
-    epc_df = pl.read_parquet(epc_path)
+    epc_df = pl.read_parquet(
+        epc_path,
+        columns=[
+            "UPRN",
+            "COUNTRY",
+            "POSTCODE",
+            "PROPERTY_TYPE",
+            "BUILT_FORM",
+            "CURRENT_ENERGY_RATING",
+        ],
+    )
+
+    logging.info("Add LSOA, MSOA, LAD, and rural-urban indicators to EPC")
+    epc_df = output_areas.standardise_col_postcode(epc_df, pcd_col="POSTCODE")
+    onspd_df = output_areas.load_transform_df_area_info()
+    epc_df = epc_df.join(onspd_df, how="left", on="POSTCODE")
 
-    # Add feature: lat/long
     logging.info("Adding lat/lon data to EPC")
     uprn_latlon_df = lat_lon.transform_df_osopen_uprn_latlon()
     epc_df = epc_df.join(uprn_latlon_df, how="left", on="UPRN")
     epc_gdf = lat_lon.generate_gdf_uprn_coords(epc_df, usecols=["UPRN", "lad_code"])
 
-    # Replace `lad_code` from postcode with `lad_code` from geospatial join and postcode
-    logging.info("Adding LAD code with geospatial join")
-    uprn_lad_df = output_areas.sjoin_df_uprn_lad_code(epc_gdf)
-    epc_df = epc_df.drop("lad_code").join(uprn_lad_df, how="left", on="UPRN")
+    # TODO this is far too slow and is only used to correct some EPC records with incorrect postcodes which get joined to the wrong LSOA/MSOA
+    # TODO can we filter the dataset somehow and only do the join with incorrect postcodes?
+    # # Replace `lad_code` from postcode with `lad_code` from geospatial join and postcode
+    # logging.info("Adding LAD code with geospatial join")
+    # uprn_lad_df = output_areas.sjoin_df_uprn_lad_code(epc_gdf)
+    # epc_df = epc_df.drop("lad_code").join(uprn_lad_df, how="left", on="UPRN")
```

Comment:
Thanks so much for writing this @lizgzil ! This wasn't something that I was planning to address in this PR because I believe it's an edge case problem that only affects a few of the EPC records*.

Because these lines of code add about 1hr+ to the run time and don't add enough value, I commented them out for the time being. 

*The original edge case was found when we were adding the conservation area feature based on LAD code and noticed that a single EPC record had an erroneous postcode and therefore had been attributed to the wrong local authority, thus being incorrectly assigned a flag that said it was in an LA without conservation area data while also being flagged as in a conservation area.

Reply:
I have added issue #95 with your suggestion to address this separately @lizgzil .